cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -654,14 +676,15 @@ public void getProgramStatusSchedules(HttpRequest request, HttpResponder respond
       }
     } else {
       // Query for schedules with all the statuses if no query status is specified
-      for (co.cask.cdap.api.ProgramStatus status: co.cask.cdap.api.ProgramStatus.values()) {
-        queryProgramStatuses.add(status);
-      }
-    }
-    for (String triggerKey : Schedulers.triggerKeysForProgramStatuses(triggerProgramId, queryProgramStatuses)) {
-      schedules.addAll(programScheduler.findSchedules(triggerKey));
+      Collections.addAll(queryProgramStatuses, co.cask.cdap.api.ProgramStatus.values());
     }
-    List<ScheduleDetail> details = Schedulers.toScheduleDetails(filterSchedulesByStatus(schedules, scheduleStatus));
+
+    List<ScheduleDetail> details = programScheduleService.findTriggeredBy(triggerProgramId, queryProgramStatuses)
+      .stream()
+      .filter(programScheduleStatus == null ?
```

Comment:
I think it's cleaner to write the lambda as `filter(r -> programScheduleStatus == null || r.getMeta().getStatus().equals(programScheduleStatus))`. Basically only one function instead of two.